site/source/docs/porting/Audio.rst

```diff
@@ -0,0 +1,143 @@
+.. _Audio:
+
+=====
+Audio
+=====
+
+Emscripten supports OpenAL 1.1 "out of the box", with no additional compilation flag required. The implementation uses the Web Audio API.
+
+You can reasonably expect ported OpenAL applications to "just work" with no additional effort.  
+However, there are some implementation-specific aspects which are worth considering - these are documented here.
+
+
+.. _Audio-openal-extensions-g:
+
+Supported OpenAL extensions
+===========================
+
+The following extensions are supported by Emscripten's OpenAL implementation.
+
+- ``ALC_SOFT_pause_device``;
+- ``ALC_SOFT_HRTF``;
+- ``AL_EXT_float32``;
+- ``AL_SOFT_loop_points``;
+- ``AL_SOFT_source_length``;
+- ``AL_EXT_source_distance_model``;
+- ``AL_SOFT_source_spatialize``;
+
+.. warning:: This doesn't mean you should just assume their presence ! For correctness, you should **always** check if an extension is supported before using it, like a good application would do.
+
+
+.. _Audio-guidelines-g:
+
+Guidelines for Audio on Emscripten
+==================================
+
+First: Do not loop on OpenAL calls.
+
+Put simply, avoid completely this kind of code :
+
+.. code-block:: c
+
+    while(nframes < THE_NUMBER_OF_FRAMES_WE_WANT)
+        alcGetIntegerv(device, ALC_CAPTURE_SAMPLES, 1, &nframes);
+
+
+The reason for this is simple: Your application needs to yield to the Javascript main loop for audio processing to take place (See :ref:`Browser main loop <emscripten-runtime-environment-main-loop>`).
+
+The above snippet usually works because most OpenAL implementations own and manage one or more separate threads. This is not the case in Emscripten.
+
+What you must do instead is perform each such query only once per "main loop iteration" (i.e the callback you provide via :c:func:`emscripten_set_main_loop` or :c:func:`emscripten_set_main_loop_arg`).
+
+
+Second: Avoid creating and destroying resources relentlessly.
```

Comment:
This second point would probably be good to be removed altogether. Referencing poor garbage collection implementations in browsers can read a bit FUD-mongering. In general we do not find OpenAL garbage collection to be a big performance problem, but `alBufferData` is the problematic one. Referring to the usual "do resource loading in cold paths" optimization would be ok however - that is also advice that is sound for native code.